gpii/node_modules/flowManager/src/SettingsPutHandler.js

```diff
@@ -86,3 +92,17 @@ gpii.flowManager.cloudBased.settings.put.handleRequest = function (that, gpiiKey
         return;
     });
 };
+
+/**
+ * This function verifies if all preference keys are in allowed list..
+ *
+ * @param {Object} prefsDetail - The object that are in the path of "contexts.gpii-defaults.preferences" of a prefs safe.
+ * @param {Array} allowedPrefs - An array of preference keys that are allowed.
+ * @return {Boolean} - If all preference keys are in the allowed list, return true. Otherwise, return false.
+ */
+gpii.flowManager.cloudBased.settings.isPrefsAllowedToWrite = function (prefsDetail, allowedPrefs) {
```

Comment:
I seem to remember that you had implemented a generalised "intersect" utility before, but perhaps it was in code that is now deleted